src/plugins/discover/public/application/main/utils/fetch_all.ts

```diff
@@ -67,7 +67,7 @@ export function fetchAll(
   const searchSource = savedSearch.searchSource.createChild();
 
   try {
-    const dataView = searchSource.getField('index')!;
+    const dataView = searchSource.getField('index');
```

Comment:
Another cleanup at several places. In search source, the index field can be `undefined`, In Discover we were saying, this is never the case. Now this is aligned. And Discover won't break when such a hostile search source comes along